core/http-auth/src/main/java/software/amazon/awssdk/http/auth/internal/util/SignerUtils.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.internal.util;
+
+import static software.amazon.awssdk.http.auth.internal.util.HttpChecksumUtils.hash;
+import static software.amazon.awssdk.http.auth.internal.util.SignerConstant.AWS4_SIGNING_ALGORITHM;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.utils.BinaryUtils;
+import software.amazon.awssdk.utils.Logger;
+
+/**
+ * Utility methods to be used by various AWS Signer implementations.
+ * This class is strictly internal and is subjected to change.
+ */
+@SdkInternalApi
+public class SignerUtils {
```

Comment:
I've marked everything as internal, such that we can re-org it later. In my opinion, http-auth should be generic (can have checksum and http-related things). I don't know why the sigv4 signers are under auth, when they should probably be under http-auth-aws. Parking this as it can be done later.